ssh/resource_resource.go

```diff
@@ -131,96 +147,30 @@ func resourceResource() *schema.Resource {
 	}
 }
 
-func resourceResourceDelete(_ context.Context, d *schema.ResourceData, _ interface{}) diag.Diagnostics {
-	var diags diag.Diagnostics
-	d.SetId("")
-	return diags
-}
-
-func resourceResourceUpdate(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+func resourceResourceDelete(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	var diags diag.Diagnostics
-	config := m.(*Config)
 
-	bastionHost := d.Get("bastion_host").(string)
-	user := d.Get("user").(string)
-	hostUser := d.Get("host_user").(string)
-	privateKey := d.Get("private_key").(string)
-	hostPrivateKey := d.Get("host_private_key").(string)
-	host := d.Get("host").(string)
-	commandsAfterFileChanges := d.Get("commands_after_file_changes").(bool)
-	agent := d.Get("agent").(bool)
-	timeout := d.Get("timeout").(string)
-	port := d.Get("port").(string)
-	bastionPort := d.Get("bastion_port").(string)
+	when := d.Get("when").(string)
 
-	timeoutValue, err := calcTimeout(timeout)
-	if err != nil {
-		return diag.FromErr(err)
+	if when == "destroy" {
+		diags = mainRun(ctx, d, m, false)
 	}
-
-	if len(hostUser) == 0 {
-		hostUser = user
-	}
-
-	if len(hostPrivateKey) == 0 {
-		hostPrivateKey = privateKey
-	}
-
-	// Collect SSH details
-	privateIP := host
-	ssh := &easyssh.MakeConfig{
-		User:   hostUser,
-		Server: privateIP,
-		Port:   port,
-		Proxy:  http.ProxyFromEnvironment,
-		Bastion: easyssh.DefaultConfig{
-			User:   user,
-			Server: bastionHost,
-			Port:   bastionPort,
-		},
-	}
-	if hostPrivateKey != "" {
-		if agent {
-			return diag.FromErr(fmt.Errorf("agent mode is enabled, not expecting a private key"))
-		}
-		ssh.Key = hostPrivateKey
-	}
-	if privateKey != "" {
-		ssh.Bastion.Key = privateKey
-	}
-
-	if d.HasChange("file") {
-		createFiles, diags := collectFilesToCreate(d)
-		if len(diags) > 0 {
-			return diags
-		}
-		if err := copyFiles(ssh, config, createFiles); err != nil {
-			return diag.FromErr(fmt.Errorf("copying files to remote: %w", err))
-		}
-		if commandsAfterFileChanges {
-			commands, diags := collectCommands(d)
-			if len(diags) > 0 {
-				return diags
-			}
-			// Run commands
-			stdout, errDiags, err := runCommands(commands, time.Duration(timeoutValue), ssh, m)
-			if err != nil {
-				return errDiags
-			}
-			_ = d.Set("result", stdout)
-		}
+	if !hasErrors(diags) {
+		d.SetId("")
 	}
```

Comment:
Does it matter if we set an ID when there is an error? This is probably a best practice but I think it's mostly a no-op.

Reply:
A successfully destroy should blank the id, at least this is a practice I've seen in all of the provider codes. However, this might indeed be a noop if TF core just looks at the diags response

Reply:
Oh I put this in the wrong place 🤦